### ISSUE TITLE
16 core CPU Instance - 16 Mbps 2x streams -> Transcoding & Thread Blocking & Memory Leak 

### ISSUE THREAD

**Issue by mekya** [ant-media/Ant-Media-Server#3129]
Environment: 16 Core CPU 2.2Ghz
Server: Ant Media Server 2.3.0

Steps to reproduce
1. Publish two streams with RTMP in 16Mbps 
2. Add 1080p adaptive bitrate and make the bitrate 4Mbps
3. Make the publish run for one day

Expected behaviour
It should continue working without restart and memory leak

Actual behaviour
It fires out of memory

**Comment by chai2** [comment]
How can we produce these memory leak issues? 

We would like to have a test and reproduce with 2.3.0 and hoping they will be resolved in 2.3.1

**Comment by mekya** [comment]
Hi @chai2 
I have entered the reproduce scenario. This is reported by one of our user and we schedule it quickly in our sprint meeting. 

In addition, this is an extreme case because transcoding 16Mbps in a CPU is not a good idea. We see lots of frame drops and exceeding the queue sizes.  GPU should be used for that. Nevertheless, we're investigating the case. 

Your case is a different than this scenario. Btw, I've shared the latest snapshot with you in the e-mail. 

Take care my friend :)

Regards,
A. Oguz
